File: CPlusPlus/ExampleTests/test_json.cpp
#include <gtest/gtest.h>
#include <iostream>
#include "common/common.h"
#include "json_glue.h"

TEST(Json, Scenario_ConvertToJson) {
    JsonGlue glue;
    std::vector<SimpleClassString> objectList1 = {
        SimpleClassString::from_vec({"1", "B"}),
    };
    glue.given_one_object_is(objectList1);

    glue.then_json_should_be("{anInt:\"1\",aString:\"B\"}");
}

TEST(Json, Scenario_ConvertFromJson) {
    JsonGlue glue;
    glue.given_json_is("{anInt:  \"1\"   ,   aString:\"B\"  }");
    std::vector<SimpleClassString> objectList2 = {
        SimpleClassString::from_vec({"1", "B"}),
    };
    glue.then_the_converted_object_is(objectList2);

}

TEST(Json, Scenario_ConvertToJsonArray) {
    JsonGlue glue;
    std::vector<SimpleClassString> objectList3 = {
        SimpleClassString::from_vec({"1", "B"}),
        SimpleClassString::from_vec({"2", "C"}),
    };
    glue.given_a_table_is(objectList3);

    glue.then_json_for_table_should_be("[ {anInt:\"1\",aString:\"B\"} \n, {anInt:\"2\",aString:\"C\"} \n]");
}

TEST(Json, Scenario_ConvertFromJsonArray) {
    JsonGlue glue;
    glue.given_json_for_table_is("[    {anInt:  \"1\"   ,   aString:\"B\"  },\n{anInt:  \"2\"   ,   aString:\"C\"  }\n]\n");
    std::vector<SimpleClassString> objectList4 = {
        SimpleClassString::from_vec({"1", "B"}),
        SimpleClassString::from_vec({"2", "C"}),
    };
    glue.then_the_converted_table_should_be(objectList4);

}

File: Python/ExampleTests/Test_Json.py
import pytest
from common import *
from json_glue import JsonGlue

def test_Scenario_ConvertToJson():
    glue = JsonGlue()

    object_list_1 = [
        SimpleClassString('1', 'B'),
    ]
    glue.given_one_object_is(object_list_1)

    glue.then_json_should_be('{anInt:"1",aString:"B"}')


def test_Scenario_ConvertFromJson():
    glue = JsonGlue()

    glue.given_json_is('{anInt:  "1"   ,   aString:"B"  }')

    object_list_2 = [
        SimpleClassString('1', 'B'),
    ]
    glue.then_the_converted_object_is(object_list_2)


def test_Scenario_ConvertToJsonArray():
    glue = JsonGlue()

    object_list_3 = [
        SimpleClassString('1', 'B'),
        SimpleClassString('2', 'C'),
    ]
    glue.given_a_table_is(object_list_3)

    glue.then_json_for_table_should_be('[ {anInt:"1",aString:"B"} \n, {anInt:"2",aString:"C"} \n]')


def test_Scenario_ConvertFromJsonArray():
    glue = JsonGlue()

    glue.given_json_for_table_is('[    {anInt:  "1"   ,   aString:"B"  },\n{anInt:  "2"   ,   aString:"C"  }\n]\n')

    object_list_4 = [
        SimpleClassString('1', 'B'),
        SimpleClassString('2', 'C'),
    ]
    glue.then_the_converted_table_should_be(object_list_4)


